### PR TITLE
More executors

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: haskell-build/haskell-build-orb.yml [hunk 1]
```diff
@@ -107,6 +107,24 @@ examples:
                   - run: echo "I run after tests"
 
 executors:
+  ghc-7.10.3:
```
Comment (AlexeyRaga) [inline haskell-build/haskell-build-orb.yml]: Do all of these images contain `cabal >= 2.2` (or where `new-configure`/`new-build` + git support appear)?

File: haskell-build/haskell-build-orb.yml [hunk 2]
```diff
@@ -107,6 +107,24 @@ examples:
...
+  ghc-8.0.2:
+    docker:
+      - image: quay.io/haskell_works/ghc-8.0.2:latest
+  ghc-8.2.2:
+    docker:
+      - image: quay.io/haskell_works/ghc-8.2.2:latest
+  ghc-8.4.4:
+    docker:
+      - image: quay.io/haskell_works/ghc-8.4.4:latest
+  ghc-8.4.3:
```
Comment (AlexeyRaga) [inline haskell-build/haskell-build-orb.yml]: Does it work with `.` in the executor name? I thought I tried and it didn't work, and that's why I updated them to `ghc-863`. But if it works for you then it must have been something else..

Comment (AlexeyRaga) [reply haskell-build/haskell-build-orb.yml]: No, it doesn't work this way:
```
$ circleci orb validate haskell-build-orb.yml
Error: ERROR IN CONFIG FILE:
[#/executors] 6 schema violations found
1. [#/executors/ghc-8.0.2] string [ghc-8.0.2] does not match pattern ^[a-z][a-z\d_-]*$
2. [#/executors/ghc-8.2.2] string [ghc-8.2.2] does not match pattern ^[a-z][a-z\d_-]*$
3. [#/executors/ghc-7.10.3] string [ghc-7.10.3] does not match pattern ^[a-z][a-z\d_-]*$
4. [#/executors/ghc-8.4.3] string [ghc-8.4.3] does not match pattern ^[a-z][a-z\d_-]*$
5. [#/executors/ghc-8.4.4] string [ghc-8.4.4] does not match pattern ^[a-z][a-z\d_-]*$
6. [#/executors/ghc-8.6.3] string [ghc-8.6.3] does not match pattern ^[a-z][a-z\d_-]*$
```

File: haskell-build/haskell-build-orb.yml (hunk 1 above)
Comment (newhoggy) [reply haskell-build/haskell-build-orb.yml]: I believe all the images have the same version of cabal.

`7.10.3` definitely works with our images: https://circleci.com/workflow-run/6adfcf96-1473-4c83-aaf6-b4f52a915e11

File: haskell-build/haskell-build-orb.yml (hunk 2 above)
Comment (newhoggy) [reply haskell-build/haskell-build-orb.yml]: That's annoying.  Can we put underscores instead?

Comment (newhoggy) [reply haskell-build/haskell-build-orb.yml]: I've updated to use underscores instead.

Comment (AlexeyRaga) [reply haskell-build/haskell-build-orb.yml]: Cool! Can you also update the "examples" section?
I see that the old `844` and `863` are still there, have you decided to keep them for compatibility reasons?

**Pull request merged**